libstuff/STCPServer.cpp

```diff
@@ -23,6 +23,17 @@ STCPServer::Port* STCPServer::openPort(const string& host) {
     return &*portIt;
 }
 
+
+void STCPServer::shutdownPorts() {
```

Comment:
Actually, I'm going to revise this. Being that this is a listening socket, and we've already accepted all new sockets coming in from it, and we're not going to read from it again, I don't think that this will really have any effect.

Instead, I won't accept any new sockets once we start shutting down. Closing the listening port with clients waiting for us to accept their connection should cause them to get an ECONNRESET, which they should be able to handle appropriately.

Then we don't need to worry about the extra `recv` call above, either, because we won't have just accepted any new sockets that may or may not have data waiting on them.